packages/sdk/odahuflow/sdk/clients/api.py

```diff
@@ -428,35 +440,20 @@ def _request(self,
         request_kwargs = self.url_builder.build_request_kwargs(url_template, payload, action, stream,
                                                                token=self.authenticator.token)
         connection_timeout = self.conn_policy.get_conn_timeout(timeout)
-        left_retries = self.conn_policy.get_left_retries()
-
-        raised_exception = None
-        while left_retries > 0:
-            try:
-                LOGGER.debug('Requesting {}'.format(self.url_builder.build_url(url_template)))
 
-                if client:
-                    response = client.request(timeout=connection_timeout, stream=stream, **request_kwargs)
-                else:
-                    response = requests.request(timeout=connection_timeout, stream=stream, **request_kwargs)
-
-                LOGGER.debug('Response status code: "{}"'.format(response.status_code))
-
-            except requests.exceptions.ConnectionError as exception:
-                LOGGER.error('Failed to connect to {}: {}. Retrying'.format(self._base_url, exception))
-                raised_exception = exception
+        try:
```

Comment:
Looks way better! :+1: Let's add the same to Async client as well?

Reply:
Async client has use aiohttp that have no transport adapter mechanism. So lt's keep it as is, 
First of all, it looks that async client catch propert exception to implement custom retry logic
And second, async client is used much less than blocking to refactoring it now